### PR TITLE
KOGITO-641: Build fails processing dtable, when no drools-decisiontables

### PR DESCRIPTION
**Pull request by evacchi**
An helpful error message is now printed when XLS are present, but the DTABLE dependency is missing

```
2019-11-26 10:56:33,120 [main] ERROR A Decision Table resource was found, but a necessary dependency is missing. 
Verify you have added decision table support to your project dependencies: 

<dependency>
      <groupId>org.kie.kogito</groupId>
      <artifactId>drools-decisiontables</artifactId>
</dependency>

```

### CODE REVIEW

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegenTest.java [hunk 1]
```diff
@@ -21,14 +21,26 @@
...
 import org.kie.api.io.ResourceType;
 import org.kie.kogito.codegen.GeneratedFile;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class IncrementalRuleCodegenTest {
 
+    @BeforeEach
+    public void blah() {
```
Comment (mswiderski) [inline kogito-codegen/src/test/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegenTest.java]: maybe not the best method name?

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/RuleCodegenError.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.kie.kogito.codegen.rules;
```
Comment (mswiderski) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/RuleCodegenError.java]: missing license header

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java [hunk 3]
```diff
@@ -182,7 +198,7 @@ public ApplicationSection section() {
 
         if (modelBuilder.hasErrors()) {
             ApplicationGenerator.logger.error( modelBuilder.getErrors().toString() );
-            return Collections.emptyList();
+            throw new RuleCodegenError();
```
Comment (mswiderski) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java]: shoudn't we pass the errors to the exception?

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegenTest.java (hunk 1 above)
Comment (evacchi) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegenTest.java]: whoopsie :P

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java (hunk 3 above)
Comment (evacchi) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java]: Yes, maybe that's a better way to do it? Currently the error just says "please scroll up to read more"

Comment (mswiderski) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java]: isn't that like duplicated error? one that has the actual info and the the other that just says... try to find it somewhere up the scroll?

**Pull request merged**